cpp/daal/src/algorithms/service_kernel_math.h

```diff
@@ -261,6 +262,105 @@ class EuclideanDistances : public PairwiseDistances<FPType, cpu>
     TArray<FPType, cpu> normBufferB;
 };
 
+template <typename FPType, CpuType cpu>
+bool solveEquationsSystemWithCholesky(FPType * a, FPType * b, size_t n, size_t nX, bool sequential)
+{
+    /* POTRF and POTRS parameters */
+    char uplo     = 'U';
+    DAAL_INT info = 0;
+
+    /* Perform L*L' factorization of A */
+    if (sequential)
+    {
+        Lapack<FPType, cpu>::xxpotrf(&uplo, (DAAL_INT *)&n, a, (DAAL_INT *)&n, &info);
+    }
+    else
+    {
+        Lapack<FPType, cpu>::xpotrf(&uplo, (DAAL_INT *)&n, a, (DAAL_INT *)&n, &info);
+    }
+    if (info != 0) return false;
+
+    /* Solve L*L' * x = b */
+    if (sequential)
+    {
+        Lapack<FPType, cpu>::xxpotrs(&uplo, (DAAL_INT *)&n, (DAAL_INT *)&nX, a, (DAAL_INT *)&n, b, (DAAL_INT *)&n, &info);
+    }
+    else
+    {
+        Lapack<FPType, cpu>::xpotrs(&uplo, (DAAL_INT *)&n, (DAAL_INT *)&nX, a, (DAAL_INT *)&n, b, (DAAL_INT *)&n, &info);
+    }
+    return (info == 0);
+}
+
+template <typename FPType, CpuType cpu>
+bool solveEquationsSystemWithPLU(FPType * a, FPType * b, size_t n, size_t nX, bool sequential)
+{
+    /* Extend symmetric matrix to generic through filling of upper triangle */
+    for (size_t i = 0; i < n; ++i)
+    {
+        for (size_t j = 0; j < i; ++j)
+        {
+            a[j * n + i] = a[i * n + j];
+        }
+    }
```

Comment:
Is it required?
I thought we have full matrix A in both regressions and ALS

Reply:
Yes, it's required. In ALS and regressions, previous operations return symmetric matrix